src/pudl/workspace/datastore.py

```diff
@@ -609,7 +593,18 @@ def main():
         logfile=args.logfile, loglevel=args.loglevel
     )
 
-    dstore = _create_datastore(args)
+    if args.pudl_in:
+        set_path_overrides(input_dir=args.pudl_in)
+
+    cache_path = None
+    if not args.bypass_local_cache:
+        cache_path = PudlPaths().input_dir
+
+    dstore = Datastore(
+        gcs_cache_path=args.gcs_cache_path,
+        sandbox=args.sandbox,
+        local_cache_path=cache_path,
+    )
```

Comment:
Does it make sense to integrate the remove cache URLs / remote output URLs into the settings object?

Reply:
Yep, this is indeed bit of a gray area IMO. What is the overall ETL configuration (inputs/outputs certainly), and what is the configuration of the individual pipeline run (e.g. where we'd configure datasets, years, perhaps output destinations). In essence, I'm assuming that PUDL_INPUT/PUDL_OUTPUT remain fixed within an environment (whether this is local or docker image) and etl configuration varies. Output destination seems like part of the etl config, perhaps also gcs cache configuration, although there we might expect more fixed setup as well.

Reply:
@jdangerx do you have thoughts on this?

Reply:
I'd suggest adding TODO here with subsequent cleanup to keep the scope of this PR from growing?